hazelcast/src/main/java/com/hazelcast/durableexecutor/impl/DurableExecutorContainer.java

```diff
@@ -17,49 +17,81 @@
 
 public class DurableExecutorContainer {
 
-    private final String name;
-
-    private final NodeEngineImpl nodeEngine;
-
-    private final ExecutionService executionService;
-
-    private final ILogger logger;
-
-    private final int partitionId;
-
     private final int durability;
+    private final int partitionId;
+    private final boolean statisticsEnabled;
 
+    private final String name;
+    private final ILogger logger;
     private final TaskRingBuffer ringBuffer;
+    private final NodeEngineImpl nodeEngine;
+    private final ExecutorStats executorStats;
+    private final ExecutionService executionService;
 
     public DurableExecutorContainer(NodeEngineImpl nodeEngine, String name, int partitionId,
-                                    int durability, TaskRingBuffer ringBuffer) {
+                                    int durability, boolean statisticsEnabled, TaskRingBuffer ringBuffer) {
         this.name = name;
         this.nodeEngine = nodeEngine;
         this.executionService = nodeEngine.getExecutionService();
         this.partitionId = partitionId;
         this.logger = nodeEngine.getLogger(DurableExecutorContainer.class);
         this.durability = durability;
         this.ringBuffer = ringBuffer;
+        this.statisticsEnabled = statisticsEnabled;
+        this.executorStats = ((DistributedDurableExecutorService) nodeEngine.getService(SERVICE_NAME)).getExecutorStats();
     }
 
     public int execute(Callable callable) {
-        int sequence = ringBuffer.add(callable);
-        TaskProcessor processor = new TaskProcessor(sequence, callable);
-        executionService.executeDurable(name, processor);
-        return sequence;
+        try {
+            int sequence = ringBuffer.add(callable);
+            TaskProcessor processor = new TaskProcessor(sequence, callable);
+            executionService.executeDurable(name, processor);
+            return sequence;
+        } catch (RejectedExecutionException e) {
+            if (statisticsEnabled) {
+                executorStats.rejectExecution(name);
+            }
+            throw ExceptionUtil.rethrow(e);
+        }
+    }
+
+    void executeAll() {
+        try {
+            TaskRingBuffer.DurableIterator iterator = ringBuffer.iterator();
+            while (iterator.hasNext()) {
+                Object item = iterator.next();
+                boolean isCallable = iterator.isTask();
+                if (!isCallable) {
+                    continue;
+                }
+                Callable callable = (Callable) item;
+                int sequence = iterator.getSequence();
+                TaskProcessor processor = new TaskProcessor(sequence, callable);
+                executionService.executeDurable(name, processor);
+            }
+        } catch (RejectedExecutionException e) {
+            if (statisticsEnabled) {
+                executorStats.rejectExecution(name);
+            }
+            throw ExceptionUtil.rethrow(e);
```

Comment:
Will this wrap the exception? Do we have tests for this case?

Reply:
removed wrapping.